app/models/user_preference.rb

```diff
@@ -59,25 +61,67 @@ def comments_sorting=(order)
     others[:comments_sorting] = order
   end
 
+  def comments_in_reverse_order?
+    comments_sorting == 'desc'
+  end
+
+  # Provide an alias to form builders
+  alias :comments_in_reverse_order :comments_in_reverse_order?
+
+  def comments_in_reverse_order=(value)
+    others[:comments_sorting] = cast_boolean(value) ? 'desc' : 'asc'
+  end
+
   def theme
     others[:theme] || OpenProject::Themes.application_theme_identifier
   end
 
-  def theme=(order)
-    others[:theme] = order
+  def theme=(identifier)
+    others[:theme] = identifier.nil? ? nil : identifier.to_sym
+  end
+
+  def canonical_time_zone
+    return if time_zone.nil?
+
+    zone = ActiveSupport::TimeZone.new(time_zone)
+    unless zone.nil?
+      zone.tzinfo.canonical_identifier
+    end
+  end
+
+  def impaired?
+    !!impaired
+  end
+
+  def warn_on_leaving_unsaved?
+    # Need to cast here as previous values were '0' / '1'
+    cast_boolean(others.fetch(:warn_on_leaving_unsaved) { true })
   end
 
   def warn_on_leaving_unsaved
-    others.fetch(:warn_on_leaving_unsaved) { '1' }
+    warn_on_leaving_unsaved?
```

Comment:
Why not use `alias` here too? (like you did above for `comments_in_reverse_order`)